File: 백준/Bronze/25304. 영수증/영수증.py
import sys
input = sys.stdin.readline








if __name__ == "__main__":
    total = int(input())
    N = int(input())

    c_num = 0
    for i in range(N):
        money, q = map(int,input().split())
        c_num += (money * q)
    
    if(c_num == total):
        print("Yes")
    else:
        print("No")

File: 백준/Bronze/25304. 영수증/영수증.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin', 'utf8').trim().split('\n');

let X = Number(input[0]);
let N = Number(input[1]);
let sum = 0;

for (let i = 0; i < N; i++) {
    const arr = input[i + 2].trim().split(' ');
    sum += Number(arr[0]) * Number(arr[1]);
}

if (sum === X) {
    console.log("Yes");
} else {
    console.log("No");
}
